File: Day01/part1.py
def main():
    with open('./input.txt') as file:
        ordered_left = []
        ordered_right = []

        for line in file:
            line = line.strip()
            tokens = line.split()
            numbers = [int(token) for token in tokens]
            left, right = numbers

            ordered_left.append(left)
            ordered_right.append(right)

    ordered_left.sort()
    ordered_right.sort()

    diffs = 0

    for i in range(0, len(ordered_left)):
        diffs += abs(ordered_left[i] - ordered_right[i])

    print(diffs)


if __name__ == '__main__':
    main()

File: Day01/part1.pl
use strict;
use warnings;
use List::Util 'sum';

open (my $file, "<", "input.txt") or die $!;

my @left_nums;
my @right_nums;

while (my $line = <$file>) {
    my @nums = split(" ", $line);
    my $left = $nums[0];
    my $right = $nums[1];

    push(@left_nums, $left);
    push (@right_nums, $right);
}

close ($file);

@left_nums = sort(@left_nums);
@right_nums = sort(@right_nums);

my @diffs = map {abs($left_nums[$_] - $right_nums[$_])} (0..$#left_nums);
my $total = sum(@diffs);

print("Total: $total\n");
